### ISSUE TITLE
problem with _update_store in case of related field on of type boolean

### ISSUE THREAD

**Issue by tfr-odoo** [odoo/odoo#4292]
Use case:
  Define a boolean related on another boolean, with already existing data.
All the record that should have True as value is ok, but all the value that should be false are empty in the database. 

When you read the data, you cannot notice anything, it's working. But when you search to your field is false in advance search, no result shows up

the problem is here 
```python
def _update_store(self, cr, f, k):
        _logger.info("storing computed values of fields.function '%s'", k)
        ss = self._columns[k]._symbol_set
        update_query = 'UPDATE "%s" SET "%s"=%s WHERE id=%%s' % (self._table, k, ss[0])
        cr.execute('select id from '+self._table)
        ids_lst = map(lambda x: x[0], cr.fetchall())
        while ids_lst:
            iids = ids_lst[:AUTOINIT_RECALCULATE_STORED_FIELDS]
            ids_lst = ids_lst[AUTOINIT_RECALCULATE_STORED_FIELDS:]
            res = f.get(cr, self, iids, k, SUPERUSER_ID, {})
            for key, val in res.items():
                if f._multi:
                    val = val[k]
                # if val is a many2one, just write the ID
                if type(val) == tuple:
                    val = val[0]
                if val is not False:  //<====================== Here
                    cr.execute(update_query, (ss[1](val), key)
```

Ok do nothing is the value is false for char, text, int etc... but please do something for the boolean field.

Present in v7 and v8

**Comment by rco-odoo** [comment]
Fixed both on 7.0 and 8.0.

**Comment by KangOl** [comment]
Ok. So no forward-port needed.
